freqtrade/exchange/__init__.py

```diff
@@ -315,18 +342,11 @@ def get_id() -> str:
     return _API.id
 
 
-def get_fee_maker() -> float:
-    return _API.fees['trading']['maker']
-
-
-def get_fee_taker() -> float:
-    return _API.fees['trading']['taker']
-
-
-def get_fee() -> float:
+def get_fee(symbol='ETH/BTC', type='', side='', amount=1,
```

Comment:
I think we should add more test-cases for this - it now has a ton of parameters, but only a very generic test not even covering half of the optional parameters.

Reply:
I think we should make `symbol` a positional argument

Reply:
>I think we should make symbol a positional argument

@gcarq Why do you think that? There's no reason fees should be affected by symbol more than any other parameter (probably even less likely). So it should be callable as simply `get_fee()` I think.

> I think we should add more test-cases for this - it now has a ton of parameters, but only a very generic test not even covering half of the optional parameters.

@xmatthias What do you should be tested specifically? Maybe exception handling for load_markets. There isn't really any logic that isn't handled by ccxt calculate_fees since all parameters are just passed there anyway, and so we would have to mock all data used in the test anyway as to not rely on the ccxt exchange data staying the same. So any unit test we write would simply be confirming that the mocked function/data is correct and would be pretty tautological, so I think we should trust that the ccxt function `calculate_fees` is correct.

Reply:
Good points on the tests.
i agree with @gcarq  however - symbol should not default to a specific pair - assuming the bot is used with a different exchange (that's why switching to ccxt makes sense, no?) with a different fee scheme  the pair used *could* make a difference - and defaulting to something could introduce bugs when not used with Binance.

Reply:
Yeah I guess I agree. I think the fee calculation/handling should be revised overall like you mentioned earlier, and as I put in the todo list. In ccxt's `calculate_fee` all parameters are required except for `taker_or_maker`. So if we want to make it truly general I think we should make all parameters required. But then the fee calculation would have the be updated to calculate the fee when both buying and selling, and checking which symbol is used and so on, and also update the Trade object to hold this information. I think that would be best long term. 

ccxt for example returns which currency the fee is subtracted from which would be really useful for the binance/bittrex fee inconsistency. So perhaps all this should be part of a "fee-fixing" PR.